### PR TITLE
message filters in search request

### PR DESCRIPTION
**Pull request by ravlio**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: query_test.go [hunk 1]
```diff
@@ -49,20 +49,42 @@ func TestClient_Search(t *testing.T) {
...
 
-	got, err := c.Search(SearchRequest{Query: text, Filters: map[string]interface{}{
-		"members": map[string][]string{
-			"$in": {user1.ID, user2.ID},
-		},
-	}})
+	t.Run("search message text", func(t *testing.T) {
+		got, err := c.Search(SearchRequest{Query: text, Filters: map[string]interface{}{
+			"members": map[string][]string{
+				"$in": {user1.ID, user2.ID},
```
Comment (pascaldekloe) [inline query_test.go]: Search for user IDs in "search message text"?

Comment (Bogdan-D) [reply query_test.go]: filters are required in any way

File: query.go [hunk 2]
```diff
@@ -105,9 +105,9 @@ func (c *Client) QueryChannels(q *QueryOption, sort ...*SortOption) ([]*Channel,
 
 type SearchRequest struct {
 	// Required
-	Query   string                 `json:"query"`
-	Filters map[string]interface{} `json:"filter_conditions"`
-
+	Query          string                 `json:"query"`
+	Filters        map[string]interface{} `json:"filter_conditions"`
+	MessageFilters map[string]interface{} `json:"message_filters"`
```
Comment (Bogdan-D) [inline query.go]: are message filters required or optional?

Comment (ravlio) [reply query.go]: good point. At least one of the fields is required. Validator.v9 has troubles with cross-field validation, so I've added checking inside the code.

File: query.go [hunk 3]
```diff
@@ -125,6 +126,9 @@ type searchMessageResponse struct {
 func (c *Client) Search(request SearchRequest) ([]*Message, error) {
 	var buf strings.Builder
 
+	if request.Query == "" && request.Filters == nil && request.MessageFilters == nil {
```
Comment (pascaldekloe) [inline query.go]: You probably want a `len(request.Filters) == 0 && len(request.MessageFilters) == 0` here. Len works with the nil value—no need for an extra check.

File: query_test.go (hunk 1 above)
Comment (Bogdan-D) [reply query_test.go]: probably members can be replaced with channel id

File: query.go (hunk 3 above)
Comment (ferhatelmas) [reply query.go]: We can lint this.

**Comment by keyneston** [comment]
@Bogdan-D is this held up just due to failed builds? Which I'm guessing is due to the cleanup issue I've otherwise fixed. If so I can rebase and merge.

**Pull request closed without merging**